script.js

```diff
@@ -1,25 +1,237 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  const episodeCards = makePageForEpisodes(allEpisodes);
-  document.body.append(...episodeCards);
+const showDropDownMenu = document.querySelector('#selectShow');
+let EpisodeDropDown = document.querySelector('#selectEpisode')
+const input = document.querySelector('#input');
+const readMore = document.querySelector('#readMore');
+const home = document.querySelector('#home');
+
+let selectedShow = 1;
+let selectedEpisode = "";
+let SearchTerm = "";
+
+// fetch Api function to fetch the shows
+async function getAllShows() {
+  return await fetch("https://api.tvmaze.com/shows").then((data) => {
+    return data.json();
+  })
+}
+// fetch Api function to fetch the Episodes of selected show
+async function getAllEpisodes() {
+  const rootElem = document.querySelector('#root');
+  const massage = `<div>Cant load the Episodes</div>`;
+
+  return await fetch(`https://api.tvmaze.com/shows/${selectedShow}/episodes`)
+    .then((response) => {
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! Status: ${response.status}`)
+      }
+      return response.json();
+
+    })
+}
+
+// function to display the all show in dropdown menu
+function displayShowList() {
+
+  getAllShows().then((showList) => {
+
+    showList.forEach((show) => {
+      const option = document.createElement('option');
+      showDropDownMenu.appendChild(option);
+      option.textContent = show.name;
+      option.setAttribute('id', `${show.id}`);
+    });
+  })
+
+  showDropDownMenu.addEventListener('change', () => {
+    let selectedOption = showDropDownMenu.options[showDropDownMenu.selectedIndex];
+    clearCard();
+    clearShows()
+    selectedShow = selectedOption.getAttribute('id');
+    makePageCards();
+  });
+
+}
+
+// function to display the all Episodes in dropdown menu of selected show
+function displayEpisodeList() {
+
+  getAllEpisodes().then((Episodes) => {
+    Episodes.forEach((ep) => {
+      const seasonPluEp = "S" + ep.season.toString().padStart(2, "0") + "E" + ep.number.toString().padStart(2, "0");
+      const option = document.createElement('option');
+      EpisodeDropDown.appendChild(option);
+      option.setAttribute('id', `${ep.id}`);
+      option.textContent = seasonPluEp + "-" + ep.name;
+    });
+  })
+
+
+  EpisodeDropDown.addEventListener('change', () => {
+
+    let selectedOption = EpisodeDropDown.options[EpisodeDropDown.selectedIndex];
+    clearCard();
+    clearShows();
+    const episodeName = selectedOption.textContent.split('-')
+    SearchTerm = episodeName[1];
+    makePageCards();
```

Comment:
Could you please elaborate how would the function `makePageCards` act in the scope of this current function?

Reply:
When I select a episode from the episode dropdown I only see a black screen, is that the intended outcome or is it a bug?

Reply:
`makePageCards` create filtered cards for Episodes acoording to `SearchTerm`. This function is Updating the SearchTerm and calling the makePageCards function to create filtered cards.


Yes, there is bug in Episodes dropDown. I will try to fix it. Thank you